sdk/core/core-amqp/src/util/utils.ts

```diff
@@ -198,12 +200,37 @@ export class Timeout {
 
 /**
  * A wrapper for setTimeout that resolves a promise after t milliseconds.
- * @param {number} t - The number of milliseconds to be delayed.
+ * @param {number} delayInMs - The number of milliseconds to be delayed.
+ * @param {AbortSignalLike} abortSignal - The abortSignal associated with containing operation.
+ * @param {string} abortErrorMsg - The abort error message associated with containing operation.
  * @param {T} value - The value to be resolved with after a timeout of t milliseconds.
  * @returns {Promise<T>} - Resolved promise
  */
-export function delay<T>(t: number, value?: T): Promise<T> {
-  return new Promise((resolve) => setTimeout(() => resolve(value), t));
+export function delay<T>(
+  delayInMs: number,
+  abortSignal?: AbortSignalLike,
+  abortErrorMsg?: string,
+  value?: T
+): Promise<T> {
+  return new Promise((resolve, reject) => {
+    if (abortSignal && abortSignal.aborted) {
+      return reject(
+        new AbortError(abortErrorMsg ? abortErrorMsg : `The delay was cancelled by the user.`)
```

Comment:
The error message related checks are written twice. We can extract the message upfront.

`abortErrorMsg  = abortErrorMsg  || "The delay has been cancelled by the user."`